### PR TITLE
Don't update R-base packages when preparing step_rcmdcheck()

### PR DESCRIPTION
**Pull request by pat-s**
#102 

PR example build: https://travis-ci.org/mlr-org/mlr3/builds/452371055

Build with current behavior: https://travis-ci.org/mlr-org/mlr3/builds/451944934

Time save: ~ 3 minutes

### CODE REVIEW

**Comment by krlmlr** [comment]
Deploy preview for *wonderful-volhard-261e08* ready!

Built with commit 41a93315c4933a37d046ba613d66ce24f315f616

https://deploy-preview-103--wonderful-volhard-261e08.netlify.com

**Comment by pat-s** [comment]
Thinking more about it, we should then exclude all default pkgs: https://cran.r-project.org/src/contrib/3.3.3/Recommended/

And we should detect if these pkgs are deps of the pkg and if so, still update them.

**Comment by pat-s** [comment]
> And we should detect if these pkgs are deps of the pkg and if so, still update them.

Or leave this feature out for now and give the responsibility to the user to update them manually?

File: R/steps-rcmdcheck.R [hunk 1]
```diff
@@ -13,7 +13,10 @@ TicStepWithPackageDeps <- R6Class(
       # (which might well be ahead of CRAN)
       # works very poorly with custom steps that are not aware
       # of this shadow library.
-      utils::update.packages(ask = FALSE)
+      installed_packages <- installed.packages()[, "Package"]
```
Comment (krlmlr) [inline R/steps-rcmdcheck.R]: Can you please disentangle?

```r
installed_packages <- ...
pkg_names <- ...
priority <- ...
pkg_to_update <- ...
remotes::update_packages()
```

File: NEWS.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # tic 0.2.13.9010
 
 - No longer using a separate library for package checks, because it causes a lot of problems with various steps which are not aware of this (#86, #88).
-
+- R-base packages are not updated when preparing `step_rcmdcheck()`.
```
Comment (krlmlr) [inline NEWS.md]: _fledge_ updates NEWS here, but don't need to change now.

**Comment by pat-s** [comment]
```r
priority
 [1] "boot"       "class"      "cluster"    "codetools"  "foreign"    "KernSmooth" "lattice"    "MASS"       "Matrix"     "mgcv"       "nlme"       "nnet"      
[13] "rpart"      "spatial"    "survival"   "base"       "boot"       "class"      "cluster"    "codetools"  "compiler"   "datasets"   "foreign"    "graphics"  
[25] "grDevices"  "grid"       "KernSmooth" "lattice"    "MASS"       "Matrix"     "methods"    "mgcv"       "nlme"       "nnet"       "parallel"   "rpart"     
[37] "spatial"    "splines"    "stats"      "stats4"     "survival"   "tcltk"      "tools"      "utils"  
```

hm, 44 pkgs. Quite a lot. We have the following options:

1. Give action to the user to update these pkgs manually, i.e. by adding an explicit `remotes::update_packages()` call to the workflow
2. auto-detect direct pkg deps and only update if this evaluates to TRUE
3. auto-detect reverse deps and only update if this evaluates to TRUE

Thinking about it I would prefer 2. atm. What do you think?

File: R/steps-rcmdcheck.R (hunk 1 above)
Comment (krlmlr) [reply R/steps-rcmdcheck.R]: The many `installed.packages()` calls still make it hard to skim through the code: 

```r
inst <- installed.packages()
...
priority <- inst[, "Priority"]
pkg_to_update <- inst[!(priority %in% ...)]
...
```

**Comment by krlmlr** [comment]
The list has dupes, so it's not that many. I'm fine with leaving base + recommended packages alone for the update process, users can update those manually if really needed (but I think this will rarely be the case). Is this step 1 on your list? Maybe I don't understand the problem here.

**Comment by pat-s** [comment]
> Is this step 1 on your list? Maybe I don't understand the problem here.

Yes it is. We should then at least document this behavior so that pkg devs do not wonder why their pkg deps are not updated (if the pkg depends on one of these pkgs).

I documented it in `?step_rcmdcheck()`. Feel free to tweak and update to your liking. Maybe we can get this merged soon :)

**Comment by pat-s** [comment]
@krlmlr 	ready from my side

**Comment by pat-s** [comment]
> Would you mind tweaking the code that determines the packages to be updated?

Ah haven't seen your latest comment about it. Gave it a new spin based on your comment.

**Pull request merged**